### PR TITLE
update FFI readme for CMake

### LINKED ISSUE #29: Update FFI readme for CMake

### PR DESCRIPTION
**Pull request by johnpryan**
fixes #29

### CODE REVIEW

File: ffi/README.md [hunk 1]
```diff
@@ -1,80 +1,39 @@
...
+## Building native libraries
 
-The C make files are (currently) written to work on a Mac. To compile the
-libraries, go to the `c/` folder in each subproject and run:
+Each sample uses [CMake][cmake] to generate a Makefile. To build the native
+library for each sample:
 
-``` shell
-make dylib
+```bash
```
Comment (dcharkes) [inline ffi/README.md]: Cool, the less instructions the better!

Does this suffice for all supported OSes? Including the visibility rules etc?

Comment (johnpryan) [reply ffi/README.md]: I think cmake can configure things like visibility rules, but I haven't tried it. cc/ @domesticmouse

Comment (domesticmouse) [reply ffi/README.md]: First up, I’m a CMake noob, I just know it is widely used by developers doing cross platform c/c++ work. Secondly, what do you mean by visibility rules?

File: ffi/README.md [hunk 2]
```diff
@@ -1,80 +1,39 @@
...
-    ``` shell
-    nmake -f Nmakefile dll
-    ```
-
-    A .dll file should be created in the parent folder. Navigate to that and run
-    the dart file.
-
-    **Only applicable if on Windows:**
-
-    The c/c++ functions should be exported or dart will not be able to lookup the function definitions.
```
Comment (dcharkes) [inline ffi/README.md]: @domesticmouse @johnpryan This is what I was referring to.

Comment (domesticmouse) [reply ffi/README.md]: I've raised (and self assigned) https://github.com/dart-lang/samples/issues/32 to confirm that this sample works as advertised on MS Windows.

Comment (dcharkes) [reply ffi/README.md]: Awesome, thanks! :+1:

**Pull request merged**